tests/support/expectations.js

```diff
@@ -10,826 +10,844 @@ module.exports = function (request, config, isServer) {
     currentLog = log;
   };
   var superagentPackage = require('superagent/package.json');
-  var superagentUserAgentHeader = isServer ? {"User-Agent": 'node-superagent/' + superagentPackage.version} : {};
+  var superagentUserAgentHeader = isServer ? {'User-Agent': 'node-superagent/' + superagentPackage.version} : {};
   var originalSetTimeout = setTimeout;
 
-  return {
+  beforeEach(function (go) {
+    // Stubbing the superagent method
+    request.Request.prototype.end = function (fn) {
+      fn(null, 'Real call done');
+    };
 
-    'setUp': function (go) {
-      // Stubbing the superagent method
-      request.Request.prototype.end = function (fn) {
-        fn(null, 'Real call done');
-      };
+    var oldSet = request.Request.prototype.set;
+    request.Request.prototype.set = function (field, value) {
+      if (typeof field === 'object') { // spy on set to collect the used arguments
+        headers = field;
+      }
 
-      var oldSet = request.Request.prototype.set;
-      request.Request.prototype.set = function (field, value) {
-        if (typeof field === 'object') { // spy on set to collect the used arguments
-          headers = field;
-        }
+      return oldSet.call(this, field, value);
+    };
 
-        return oldSet.call(this, field, value);
-      };
-
-      // Stub setTimeout
-      Object.defineProperty(global, 'setTimeout', {
-        value: function(callbackFunc, timeout) {
-          if (!global.setTimeout.calls) {
-            global.setTimeout.calls = [];
-          }
-          global.setTimeout.calls.push([callbackFunc, timeout]); // spy on calls made to setTimeout
-          callbackFunc();
+    // Stub setTimeout
+    Object.defineProperty(global, 'setTimeout', {
```

Comment:
We should use jest's timers mocks : https://facebook.github.io/jest/docs/jest-object.html#jestrunonlypendingtimers